libvips/foreign/vips2tiff.c

```diff
@@ -855,13 +855,19 @@ wtiff_write_header(Wtiff *wtiff, Layer *layer)
 		TIFFSetField(tif, TIFFTAG_WEBP_LOSSLESS, wtiff->lossless);
 	}
 	if (wtiff->compression == COMPRESSION_ZSTD) {
-		TIFFSetField(tif, TIFFTAG_ZSTD_LEVEL, wtiff->level);
+		// Set zstd compression level - only accept valid values (1-22)
+		if ((wtiff->level>=1) && (wtiff->level<=22))
+			TIFFSetField(tif, TIFFTAG_ZSTD_LEVEL, wtiff->level);
```

Comment:
How about removing the if() and just writing:

```C
TIFFSetField(tif, TIFFTAG_ZSTD_LEVEL, VIPS_CLIP(1, wtiff->level, 22));
```

Probably better than silently ignoring out of range values.